src/coreclr/vm/comutilnative.cpp

```diff
@@ -1815,6 +1815,15 @@ FCIMPL1(UINT32, MethodTableNative::GetNumInstanceFieldBytes, MethodTable* mt)
 }
 FCIMPLEND
 
+FCIMPL1(CorElementType, MethodTableNative::GetVerifierCorElementType, MethodTable* mt)
+{
+    FCALL_CONTRACT;
+    // MethodTable::GetInternalCorElementType has unnecessary overhead for primitives and enums
+    // Call EEClass::GetInternalCorElementType directly to avoid it
+    return mt->GetClass()->GetInternalCorElementType();
```

Comment:
```suggestion
    return mt->GetInternalCorElementType();
```

I would just call the one on `MethodTable`. The overhead in there will be difficult to observe, but finding this call doesn't handle all types will be a pain to debug.

Reply:
This has impact on performance of enum.
If the element type is not set for every `MethodTable`, I would add an assert here.

Reply:
> This has impact on performance of enum.

Are you basing this on the comment that is likely more than 20 years old or because it was measured recently?

Reply:
This comment is ~4 years old and it was added as part of a fix for performance regression identified by the perf lab: https://github.com/dotnet/runtime/pull/37845 . Changing this back to `mt->GetInternalCorElementType()` would revert the fix and re-introduce the perf regression.

Reply:
Okay.